lib/pages/chat_list/chat_list_view_builder.dart

```diff
@@ -16,44 +16,50 @@ class ChatListViewBuilder extends StatelessWidget {
-                context,
-                rooms[index],
-              ),
-              onLongPress: () => controller.onLongPressChatListItem(
-                rooms[index],
-              ),
-              checkBoxWidget: ValueListenableBuilder(
-                valueListenable: controller.conversationSelectionNotifier,
-                builder: (context, conversationSelection, __) {
-                  final conversation = conversationSelection.firstWhereOrNull(
-                    (conversation) =>
-                        conversation.roomId.contains(rooms[index].id),
-                  );
-                  return Checkbox(
-                    value: conversation?.isSelected == true,
-                    onChanged: (_) {
-                      controller.toggleSelection(rooms[index].id);
+    return ValueListenableBuilder(
+      valueListenable: controller.widget.activeRoomIdNotifier,
+      builder: (context, activeRoomId, child) {
+        return ListView.builder(
+          shrinkWrap: true,
+          physics: const NeverScrollableScrollPhysics(),
+          itemCount: rooms.length,
+          itemBuilder: (BuildContext context, int index) {
+            return ValueListenableBuilder(
+              valueListenable: controller.selectModeNotifier,
+              builder: (context, _, __) {
+                return ChatListItem(
+                  rooms[index],
+                  key: Key('chat_list_item_${rooms[index].id}'),
+                  isEnableSelectMode: controller.isSelectMode,
+                  onTap: controller.isSelectMode
+                      ? () => controller.toggleSelection(rooms[index].id)
+                      : null,
+                  onSecondaryTap: () => controller.handleContextMenuAction(
+                    context,
+                    rooms[index],
+                  ),
+                  onLongPress: () => controller.onLongPressChatListItem(
+                    rooms[index],
+                  ),
+                  checkBoxWidget: ValueListenableBuilder(
+                    valueListenable: controller.conversationSelectionNotifier,
+                    builder: (context, conversationSelection, __) {
+                      final conversation =
+                          conversationSelection.firstWhereOrNull(
+                        (conversation) =>
+                            conversation.roomId.contains(rooms[index].id),
+                      );
+                      return Checkbox(
+                        value: conversation?.isSelected == true,
+                        onChanged: (_) {
+                          controller.toggleSelection(rooms[index].id);
+                        },
+                      );
                     },
-                  );
-                },
-              ),
-              activeChat: controller.activeRoomId == rooms[index].id,
+                  ),
+                  activeChat: activeRoomId == rooms[index].id,
```

Comment:
IMO: Just rebuild for `ChatListItem` no need rebuild all ListView

Reply:
done